src/fs.test.ts

```diff
@@ -0,0 +1,686 @@
+import fs from 'fs';
+import { when } from 'jest-when';
+import os from 'os';
+import path from 'path';
+import util from 'util';
+
+import {
+  createSandbox,
+  directoryExists,
+  ensureDirectoryStructureExists,
+  fileExists,
+  forceRemove,
+  readFile,
+  readJsonFile,
+  writeFile,
+  writeJsonFile,
+} from './fs';
+
+const { withinSandbox } = createSandbox('utils');
+
+describe('fs', () => {
+  describe('readFile', () => {
+    it('reads the contents of the given file as a UTF-8-encoded string', async () => {
+      await withinSandbox(async (sandbox) => {
```

Comment:
Since most of these tests are nested in a function, it would be nice to add `expect.assertions` to each of them.

Reply:
Ah, good call. I forgot that was a thing. 

Added here: [0f0417f](https://github.com/MetaMask/utils/pull/148/commits/0f0417f3f1d506e4b50cc21a5c73e3dda5a7673e)